File: epi_judge_java/epi/CountBits.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTestHandler;

public class CountBits {
  @EpiTest(testfile = "count_bits.tsv")

  public static short countBits(int x) {
    // Implement this placeholder.
    short count = 0;
    while (x > 0) {
      count += 1;
      x = x & x - 1;
    }
    return count;
  }

  public static void main(String[] args) {
    GenericTestHandler.executeTestsByAnnotation(
        new Object() {}.getClass().getEnclosingClass(), args);
  }
}

File: epi_judge_cpp/count_bits.cc

short CountBits(unsigned int x) {
  // Implement this placeholder.
  short count = 0;
  while ( x > 0 ) {
	count++;
	x &= x--;
  }
  return count;
}

#include "test_framework/test_utils_generic_main.h"

int main(int argc, char* argv[]) {
  std::vector<std::string> param_names{"x"};
  generic_test_main(argc, argv, param_names, "count_bits.tsv", &CountBits);
  return 0;
}
